docs/httpproxy.md

```diff
@@ -750,7 +750,172 @@ spec:
 
 ## HTTPProxy inclusion
 
-TODO(youngnick) document inclusion
+HTTPProxy permits the splitting of a system's configuration into separate HTTPProxy instances using **inclusion**.
+
+Inclusion, as the name implies, allows for one HTTPProxy object to be included in another, optionally with some conditions around the inclusion.
+Contour reads the inclusion tree and merges the included objects into one big object internally before rendering Envoy config.
+Importantly, the included HTTPProxy objects do not have to be in the same namespace, so this is functionally the same as the old delegation feature of the now-deprecated IngressRoute.
+
+The "root" HTTPProxy is the only entry point for a proxied virtual host and is used as the top level configuration of a cluster's proxy resources.
+Each root HTTPProxy defines a `virtualhost` key, which describes properties such as the fully qualified name of the virtual host, TLS configuration, etc.
+
+HTTPProxies that will be included do not contain  virtual host information, as they will inherit the properties of the parent HTTPProxy via `conditions`.
+This permits the owner of an HTTPProxy root to allow the inclusion of a portion of the route space inside a virtual host, and to allow that route space to be further subdivided with inclusions.
+Because the path is not necessarily used as the only key, the route space can be multi-dimensional.
+
+### Conditions and Inclusion
+
+Inclusions may be specified with conditions.
+These conditions are added to any conditions in the included HTTPProxy.
+This process repeated for any further inclusions as well.
+
+All [Conditions](#conditions) are supported for use with inclusion.
+The `prefix` condition is slightly special in that, when used for inclusion, any routes will have the value of `prefix` prepended to *their* prefix value.
+Header conditions are added to the conditions array.
+
+### Configuring inclusion
+
+Inclusion is a top-level part of the HTTPProxy `spec` element.
+It requires two fields, `name` and `namespace`, with the option of including a `conditions` block.
```

Comment:
nit: namespace isn't required. Maybe it is, probably worth checking.

Reply:
It's definitely required at the moment. I don't think it should be, but wanted to document current state.